From b06eecd9a0be064e727764142e2e70cf65c803c9 Mon Sep 17 00:00:00 2001
Message-Id: <b06eecd9a0be064e727764142e2e70cf65c803c9.1413837856.git.chang-joon.lee@intel.com>
In-Reply-To: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
References: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
From: koushik <koushikx.biswas@intel.com>
Date: Wed, 31 Jul 2013 12:17:40 +0530
Subject: [PATCH 004/312] MUST_UPSTREAM [VPG]: drm/i915: BUN vol4g[DevHSW]

WW43 2012 - DDI buffer translation corrections
WW36 2012 - Added HDMI voltage swing (not implemented
                for HDMI)

Added comments with voltage swing, pre-emphasis,
transition and non-transition values in form of table
for reference. This values are applicable only for HSW
platform.

Change-Id: I59273502598d43e268f0101048eb07c26fe34a19
Signed-off-by: koushik <koushikx.biswas@intel.com>
(cherry picked from commit 8eed6ed3db1f6ec19f298fe6c9e2728fabbc05ad)

Signed-off-by: Jon Bloomfield <jon.bloomfield@intel.com>
---
 drivers/gpu/drm/i915/intel_ddi.c |   31 +++++++++++++++++++++++++++++++
 drivers/gpu/drm/i915/intel_dp.c  |   32 ++++++++++++++++++++++++++++++++
 2 files changed, 63 insertions(+)

diff --git a/drivers/gpu/drm/i915/intel_ddi.c b/drivers/gpu/drm/i915/intel_ddi.c
index b17b9c7..3b301bb 100644
--- a/drivers/gpu/drm/i915/intel_ddi.c
+++ b/drivers/gpu/drm/i915/intel_ddi.c
@@ -211,6 +211,37 @@ void intel_prepare_ddi(struct drm_device *dev)
 		intel_prepare_ddi_buffers(dev, port);
 }
 
+
+/* Updating the new table in comments as it doesn’t cause any logic change */
+
+/************* For HSW Voltage swing levels ***************************/
+/**********************************************************************/
+/*____________________________________________________________________*/
+/*|Entry|Voltage|Pre-emphasis|Non-Transition|Transition |Pre-emphasis|*/
+/*|     |Swing  |level       |mV diff p-p   |mV diff p-p|db          |*/
+/*|------------------------------------------------------------------|*/
+/*|0    |0      |0           |400           |400        |0           |*/
+/*|------------------------------------------------------------------|*/
+/*|1    |0      |1           |400           |600        |3.5         |*/
+/*|------------------------------------------------------------------|*/
+/*|2    |0      |2           |400           |800        |6           |*/
+/*|------------------------------------------------------------------|*/
+/*|3    |0      |3           |400           |1000       |8           |*/
+/*|------------------------------------------------------------------|*/
+/*|4    |1      |0           |600           |600        |0           |*/
+/*|------------------------------------------------------------------|*/
+/*|5    |1      |1           |600           |900        |3.5         |*/
+/*|------------------------------------------------------------------|*/
+/*|6    |1      |2           |600           |1000       |4.5         |*/
+/*|------------------------------------------------------------------|*/
+/*|7    |2      |0           |800           |800        |0           |*/
+/*|------------------------------------------------------------------|*/
+/*|8    |2      |1           |1000          |1000       |2           |*/
+/*|------------------------------------------------------------------|*/
+/*|9    |       Entry 9 is only used for HDMI and DVI                |*/
+/*|------------------------------------------------------------------|*/
+/**********************************************************************/
+
 static const long hsw_ddi_buf_ctl_values[] = {
 	DDI_BUF_EMP_400MV_0DB_HSW,
 	DDI_BUF_EMP_400MV_3_5DB_HSW,
diff --git a/drivers/gpu/drm/i915/intel_dp.c b/drivers/gpu/drm/i915/intel_dp.c
index 99f033f..ce7097f 100644
--- a/drivers/gpu/drm/i915/intel_dp.c
+++ b/drivers/gpu/drm/i915/intel_dp.c
@@ -2632,6 +2632,38 @@ intel_gen7_edp_signal_levels(uint8_t train_set)
 	}
 }
 
+
+/* Updating the new table in comments as it doesn’t cause any logic change */
+
+/************* For HSW Voltage swing levels ***************************/
+/**********************************************************************/
+/*____________________________________________________________________*/
+/*|Entry|Voltage|Pre-emphasis|Non-Transition|Transition |Pre-emphasis|*/
+/*|     |Swing  |level       |mV diff p-p   |mV diff p-p|db          |*/
+/*|------------------------------------------------------------------|*/
+/*|0    |0      |0           |400           |400        |0           |*/
+/*|------------------------------------------------------------------|*/
+/*|1    |0      |1           |400           |600        |3.5         |*/
+/*|------------------------------------------------------------------|*/
+/*|2    |0      |2           |400           |800        |6           |*/
+/*|------------------------------------------------------------------|*/
+/*|3    |0      |3           |400           |1000       |8           |*/
+/*|------------------------------------------------------------------|*/
+/*|4    |1      |0           |600           |600        |0           |*/
+/*|------------------------------------------------------------------|*/
+/*|5    |1      |1           |600           |900        |3.5         |*/
+/*|------------------------------------------------------------------|*/
+/*|6    |1      |2           |600           |1000       |4.5         |*/
+/*|------------------------------------------------------------------|*/
+/*|7    |2      |0           |800           |800        |0           |*/
+/*|------------------------------------------------------------------|*/
+/*|8    |2      |1           |1000          |1000       |2           |*/
+/*|------------------------------------------------------------------|*/
+/*|9    |       Entry 9 is only used for HDMI and DVI                |*/
+/*|------------------------------------------------------------------|*/
+/**********************************************************************/
+
+
 /* Gen7.5's (HSW) DP voltage swing and pre-emphasis control */
 static uint32_t
 intel_hsw_signal_levels(uint8_t train_set)
-- 
1.7.9.5

